packages/npm/@amazeelabs/silverback-playwright/src/drupal-auth.ts

```diff
@@ -0,0 +1,13 @@
+import { PlaywrightTestArgs } from '@playwright/test';
+
+import { drupal } from './constants';
+
+export const drupalLogin = async (page: PlaywrightTestArgs['page']) => {
+  await page.goto(
+    `${drupal.baseUrl}/test-session/set?X-TEST-SESSION-USER=${drupal.adminUser.login}`,
+  );
```

Comment:
We should try to get rid of `test-session` and use playwrights `auth` capabilities instead, since we won't have to fiddle with Drupal internals and potentially create situations where it behaves differently when under test.

Reply:
> Also, I tried to use https://playwright.dev/docs/auth/#reuse-authentication-in-playwright-test. It worked but did not make a lot of sense because in the tests I needed both authorized and anonymous contexts.

Wouldn't we achieve that with different page contexts?

Reply:
When I tried the "reuse auth" feature, it worked very strange. I tried to create a new context+page in a test.

The documentation of [`browser.newContext()`](https://playwright.dev/docs/api/class-browser#browser-new-context) says:

> Creates a new browser context. It won't share cookies/cache with other browser contexts.

But for me the new page still had Drupal session cookies 🤷 

Maybe we could try together at some point?